### PR TITLE
Add batched version MultiGet API

### PR DESCRIPTION
**Pull request by yhchiang-sol**
#### Problem

RocksDB has an optimized version of MultiGet API that handle the read requests in batches so that multiple gets can be satisfied by the same file read.  But the feature is available in neither C nor rust API.

#### What's in this PR
The batched version of MultiGet for C API is available in [rocksdb-9952](https://github.com/facebook/rocksdb/pull/9952).  Once it has been merged, it should be available in the next RocksDB release.

This PR includes the draft version of the batched MultiGet for rust API.  Once the above PR is merged, I will turn this PR into ready for review.

This PR depends on https://github.com/rust-rocksdb/rust-rocksdb/pull/645 that upgrades rocksdb to v7.3.1.

### CODE REVIEW

**Comment by yhchiang-sol** [comment]
https://github.com/facebook/rocksdb/pull/9952 has been merged as it is available in the recent rocksdb release.

@dignifiedquire this PR requires rocksdb v7.3.1.  Can I know how did you generate https://github.com/rust-rocksdb/rust-rocksdb/pull/626 that upgrades the librocksdb to v7.1.2?

**Comment by yhchiang-sol** [comment]
Rebase on top of https://github.com/rust-rocksdb/rust-rocksdb/pull/645

**Comment by dignifiedquire** [comment]
> @dignifiedquire this PR requires rocksdb v7.3.1. Can I know how did you generate #626 that upgrades the librocksdb to v7.1.2?

by hand

**Review by stanislav-tkach — APPROVED** [review]

**Comment by yhchiang-sol** [comment]
Thank you, @stanislav-tkach!

Since this PR requires a newer version of rocksdb, can you also help review its parent PR https://github.com/rust-rocksdb/rust-rocksdb/pull/645 that upgrades the rocksdb version to v7.3.1?

**Comment by stanislav-tkach** [comment]
I have checked this pull request and it looks good to me except I don't really understand the changes in the `build_version.cc` file. 🙃

**Comment by yhchiang-sol** [comment]
> except I don't really understand the changes in the build_version.cc file

The build_verion.cc is actually generated by building rocksdb.  It is generated from build_version.cc.in.

Since https://github.com/facebook/rocksdb/pull/7949 updates build_version.cc.in, the resulting build_verion.cc will also reflect the change.

**Comment by yhchiang-sol** [comment]
Hello @aleksuss, our Solana needs this PR and https://github.com/rust-rocksdb/rust-rocksdb/pull/645 in order to use the optimized version of multi-get to speed up our read requests.  Both PRs have been waiting for your review for around a week.  Do you mind reviewing https://github.com/rust-rocksdb/rust-rocksdb/pull/645 that upgrades rocksdb to v7.3.1 and this PR that adds the optimized version of multi-get api?  Thank you so much!

**Comment by yhchiang-sol** [comment]
Hello @stanislav-tkach. While the PR has one approval, it looks like I am not able to merge it.

If the PR looks good to you, do you mind helping me merge the PR? Thank you!

File: src/db.rs
Comment (yhchiang-sol) [reply src/db.rs:1050]: Thank you for your time! Will do!

**Review by aleksuss — APPROVED** [review]

**Pull request merged**
